src/lib/operators/sort.cpp

```diff
@@ -200,41 +243,72 @@ class Sort::SortImpl : public AbstractReadOnlyOperatorImpl {
       }
     }
 
-    // 3. Materialization of the result: We take the sorted ValueRowID Vector, create chunks fill them until they are
-    // full and create the next one. Each chunk is filled row by row.
-    auto materialization = std::make_shared<SortImplMaterializeOutput<SortColumnType>>(_table_in, _row_id_value_vector,
-                                                                                       _output_chunk_size);
-    auto output = materialization->execute();
-
-    const auto chunk_count = output->chunk_count();
-    for (auto chunk_id = ChunkID{0}; chunk_id < chunk_count; ++chunk_id) {
-      output->get_chunk(chunk_id)->set_ordered_by(std::make_pair(_column_id, _order_by_mode));
-    }
-
-    return output;
+    return _row_id_value_vector;
   }
 
+ protected:
   // completely materializes the sort column to create a vector of RowID-Value pairs
-  void _materialize_sort_column() {
+  void _materialize_sort_column(const std::shared_ptr<PosList>& pos_list) {
```

Comment:
It might be easier to understand the effects of this method if, instead of modifying the object, it would be static and simply returned the materialized column. While it was this way before, changing this should only take a handful of lines.